contracts/ReversibleICO.sol

```diff
@@ -631,87 +622,108 @@ contract ReversibleICO is IERC777Recipient {
         returns ( bool )
     {
         Participant storage ParticipantRecord = ParticipantsByAddress[participantAddress];
-
         if( ParticipantRecord.received > 0 && ParticipantRecord.received > ParticipantRecord.returned ) {
             return true;
         }
-
-        /*
-        // ParticipantRecord.amount_accepted only available after whitelisting, so we need to check all contributions
-        uint256 ParticipantAvailableEth = 0;
-        for( uint16 i = 0; i < ParticipantRecord.contributionsCount; i++ ) {
-            Contribution storage ContributionRecord = ParticipantRecord.contributions[i];
-            if(ContributionRecord.state == uint8(ContributionStates.NOT_PROCESSED)) {
-                ParticipantAvailableEth += ContributionRecord.value;
-            }
-        }
-
-        if(ParticipantAvailableEth > 0 && ParticipantAvailableEth <= ParticipantRecord.contributed_amount) {
-            return true;
-        }
-        */
         return false;
     }
 
     /*
     *   Withdraw ( ERC777TokensRecipient method )
     */
-    function withdraw(address _from, uint256 _token_amount) internal returns (bool) {
+    function withdraw(address _from, uint256 _returned_token_amount) internal {
 
         // Whitelisted contributor sends tokens back to the RICO contract
         // - unlinke cancel() method, this allows variable amounts.
```

Comment:
why making a difference between `cancel` and `withdraw` ?
Also spelling mistake: `unlinke`